robolectric-shadows/shadows-core/src/main/resources/org/robolectric/fakes/RoboWebSettings.java.vm

```diff
@@ -602,24 +516,117 @@ public class RoboWebSettings extends WebSettings {
   public void setEnableSmoothTransition(boolean enableSmoothTransition) {
     this.enableSmoothTransition = enableSmoothTransition;
   }
+  
+  @Override
+  public void setAllowContentAccess(boolean allow) {
+    allowContentAccess = allow;
+  }
 
   @Override
-  public boolean supportZoom() {
-    return supportZoom;
+  public boolean getAllowContentAccess() {
+    return allowContentAccess;
   }
 
-  private boolean navDump;
+  @Override
+  public void setDisplayZoomControls(boolean enabled) {
+    displayZoomControls = enabled;
+  }
 
   @Override
-  public boolean getNavDump() {
-    return navDump;
+  public boolean getDisplayZoomControls() {
+    return displayZoomControls;
   }
+  // End API 11
+#end
 
+#if ($api >= 14)
+  private int textZoom = 100;
+  
   @Override
-  public void setNavDump(boolean navDump) {
-    this.navDump = navDump;
+  public int getTextZoom() {
+    return textZoom;
+  }
+
+  @Override
+  public void setTextZoom(int textZoom) {
+    this.textZoom = textZoom;
+  }
+  // End API 14
+#end
+
+#if ($api >= 16)
+  private boolean allowFileAccessFromFile = true;
+  private boolean allowUniversalAccessFromFile = true;
+
+  @Override
+  public boolean getAllowFileAccessFromFileURLs() {
+    return allowFileAccessFromFile;
+  }
+
+  @Override
+  public void setAllowFileAccessFromFileURLs(boolean allow) {
+    allowFileAccessFromFile = allow;
+  }
+
+  @Override
+  public boolean getAllowUniversalAccessFromFileURLs() {
+    return allowUniversalAccessFromFile;
+  }
+
+  @Override
+  public void setAllowUniversalAccessFromFileURLs(boolean allow) {
+    allowUniversalAccessFromFile = allow;
+  }
+  //End API 16
+#end
+
+#if ($api >= 17)
+  private boolean mediaPlaybackRequiresUserGesture = true;
+
+  @Override
+  public boolean getMediaPlaybackRequiresUserGesture() {
+    return mediaPlaybackRequiresUserGesture;
+  }
+
+  @Override
+  public void setMediaPlaybackRequiresUserGesture(boolean require) {
+    mediaPlaybackRequiresUserGesture = require;
+  }
+  //End API 17
+#end
+
+#if ($api >= 21)
+  private int mixedContentMode;
+  private boolean acceptThirdPartyCookies;  
+
+  @Override
+  public void setMixedContentMode(int mixedContentMode) {
+    this.mixedContentMode = mixedContentMode;
+  }
+
+  @Override
+  public int getMixedContentMode() {
+    return mixedContentMode;
   }
 
-  // End API 22
+  @Override
+  public void setVideoOverlayForEmbeddedEncryptedVideoEnabled(boolean b) {
```

Comment:
I guess this should ideally change an actual flag that is returned in `getVideoOverlayForEmbeddedEncryptedVideoEnabled()`.